orquesta/conducting.py

```diff
@@ -882,3 +885,40 @@ def get_task_transition_contexts(self, task_id, route):
                 contexts[task_transition_id] = self.get_task_initial_context(t[1], route)
 
         return contexts
+
+    def request_rerun_workflow_status(self, status, rerun_options):
+        # Reset workflow status.
+        current_status = self.get_workflow_status()
+        if current_status != statuses.FAILED:
+            LOG.debug('Workflow execution is not rerunable because its status is not Failed.')
+            return
+
+        self.workflow_state.status = statuses.UNSET
+
+        current_status = self.get_workflow_status()
+
+        # Create an event for the request.
+        wf_ex_event = events.WorkflowExecutionEvent(status)
+
+        sequence = self.workflow_state.get_sequence()
+        # Create a new entry in staging for the next task.
+        for i in range(len(sequence)):
+            if sequence[i]['status'] == statuses.FAILED and sequence[i]['id'] in rerun_options:
+                sequence[i]['status'] = None
+                ctxs, route = [0], 0
+                self._workflow_state.add_staged_task(
+                    sequence[i]['id'],
+                    route,
+                    ctxs=ctxs,
+                    ready=True
+                )
```

Comment:
Please don't modify the existing task entries in sequence. These tasks are a record of what happened.  So even though the tasks failed, we should keep an entry there as is.  Can you just add the new task entry in staging so the conductor returns it in `get_next_tasks`?

Reply:
Without change existing task entries in sequence for its status, later state machine `update_task_state` can't change a task status from `failed` to `running`. Maybe I misunderstand your comment.  

Fixed all other comments.

Reply:
Ok. I need to revisit this in the conductor. I don't want the existing task entry in sequence to be modified because it's a record. So let me see what needs to change. I will respond Monday.

Reply:
Ok. I briefly reviewed conductor.  So by default, `sequence` in WorkflowState allows more than one task entries for the same task id (or task name).  There's is a pointer table called `tasks` in WorkflowState that points to the latest task entry in `sequence`.  This is almost like a self-looping task.  So can you create a new task entry in sequence (i.e. https://github.com/StackStorm/orquesta/blob/master/orquesta/conducting.py#L645). If you use add_task_state function, it will also update the pointer.  You don't need to create an entry in staging in this case.  Let me know if you need me to clarify more.

Reply:
A task is added to sequence table, but the task is never staged.